### ISSUE TITLE
tty device unit times out when firstboot.service prompt remains open for too long

### ISSUE THREAD

**Issue by daandemeyer** [systemd/systemd#27731]
### systemd version the issue has been seen with

main

### Used distribution

Fedora 38

### Linux kernel version used

default Fedora kernel

### CPU architectures issue was seen on

x86_64

### Unexpected behaviour you saw

When systemd-firstboot.service prompts for settings and the prompt remains open for an extended amount of time, dev-ttyS0.device times out. This happens because serial-getty@ttyS0.service has BindsTo=dev-ttyS0.device which means the device unit is pulled in. However, because systemd-firstboot.service delays systemd-sysusers.service and systemd-sysusers.service delays systemd-udevd.service, the device unit will time out because udevd doesn't start in time to satisfy it.

One possible fix could be to add an `After=systemd-udevd.service` ordering dep on all device units.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
[   15.893777] systemd[1]: dev-ttyS0.device: Job dev-ttyS0.device/start timed out.
[   15.896096] systemd[1]: Timed out waiting for device dev-ttyS0.device.
[ TIME ] Timed out waiting for device dev-ttyS0.device.
[   15.898805] systemd[1]: Dependency failed for serial-getty@ttyS0.service.
```

**Comment by poettering** [comment]
so, the timeout for those devices is set to 10s. which is really wweird, and i don't grok where that comes from.

**Comment by DaanDeMeyer** [comment]
This was fixed by #27750
